[FLINK-21407][doc][formats] Add formats to DataStream connectors doc (#17640)
apache/flink: docs/content/docs/connectors/datastream/formats/azure_table_storage.md

---
title:  "Microsoft Azure table"
weight: 4
type: docs
aliases:
- /dev/connectors/formats/azure_table_storage.html
- /apis/streaming/connectors/formats/azure_table_storage.html
---
<!--
Licensed to the Apache Software Foundation (ASF) under one
or more contributor license agreements.  See the NOTICE file
distributed with this work for additional information
regarding copyright ownership.  The ASF licenses this file
to you under the Apache License, Version 2.0 (the
"License"); you may not use this file except in compliance
with the License.  You may obtain a copy of the License at

  http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing,
software distributed under the License is distributed on an
"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
KIND, either express or implied.  See the License for the
specific language governing permissions and limitations
under the License.
-->

# Microsoft Azure Table Storage format

> _Note: This example works starting from Flink 0.6-incubating_

Reviewer: I don't think we need to include this note, since we don't support Flink 0.6 anymore (and the documentation is specifically targeted towards Flink 1.15)

Author: agree, it was on dataSet conncetor page (cf my general comment) I did not clean anything. Will do thx for pointing out

This example is using the `HadoopInputFormat` wrapper to use an existing Hadoop input format implementation for accessing [Azure's Table Storage](https://azure.microsoft.com/en-us/documentation/articles/storage-introduction/).

1. Download and compile the `azure-tables-hadoop` project. The input format developed by the project is not yet available in Maven Central, therefore, we have to build the project ourselves.
   Execute the following commands:

```bash
git clone https://github.com/mooso/azure-tables-hadoop.git
cd azure-tables-hadoop
mvn clean install
```

2. Setup a new Flink project using the quickstarts:

```bash
curl https://flink.apache.org/q/quickstart.sh | bash
```

3. Add the following dependencies (in the `<dependencies>` section) to your `pom.xml` file:

```xml
<dependency>
   <groupId>org.apache.flink</groupId>
   <artifactId>flink-hadoop-compatibility{{< scala_version >}}</artifactId>
   <version>{{< version >}}</version>
</dependency>
<dependency>
 <groupId>com.microsoft.hadoop</groupId>
 <artifactId>microsoft-hadoop-azure</artifactId>

>  <version>0.0.4</version>

Reviewer: Nit: the ident is slightly different from the one above.

</dependency>
```

`flink-hadoop-compatibility` is a Flink package that provides the Hadoop input format wrappers.
`microsoft-hadoop-azure` is adding the project we've build before to our project.

> The project is now prepared for starting to code. We recommend to import the project into an IDE, such as Eclipse or IntelliJ. (Import as a Maven project!).

Reviewer: ```suggestion
The project is now ready for starting to code. We recommend to import the project into an IDE, such as IntelliJ. You should import it as a Maven project.
```

> Browse to the code of the `Job.java` file. Its an empty skeleton for a Flink job.

Reviewer: ```suggestion
Browse to the file `Job.java`. This is an empty skeleton for a Flink job.
```

> Paste the following code into it:

Reviewer: ```suggestion
Paste the following code:
```
